ksqldb-serde/src/main/java/io/confluent/ksql/serde/SchemaFullNameAppender.java

```diff
@@ -65,14 +79,14 @@ public String name() {
     }
   }
 
-  private static String avroCompatibleFieldName(final Field field) {
+  private static String compatibleFieldName(final Field field) {
     // Currently the only incompatible field names expected are fully qualified
     // column identifiers. Once quoted identifier support is introduced we will
     // need to implement something more generic here.
     return field.name().replace(".", "_");
```

Comment:
I'm not sure if this replacement is specific for avro format? If protobuf supports ".", we don't need to replace? In general, I'm not sure how many of the conversions are also needed for protobuf and concerned if we convert too much and break existing protobuf.

Also for the `MAP_KEY_NAME` and and `MAP_VALUE_NAME`, not sure if protobuf needs them

Reply:
It is also for Protobuf. See the bug https://github.com/confluentinc/ksql/issues/9005

Reply:
There are docs for:
Avro https://avro.apache.org/docs/current/spec.html#names
```
Names

Record, enums and fixed are named types. Each has a fullname that is composed of two parts; a name and a namespace. Equality of names is defined on the fullname.

The name portion of a fullname, record field names, and enum symbols must:

    start with [A-Za-z_]
    subsequently contain only [A-Za-z0-9_]

```
- Proto https://developers.google.com/protocol-buffers/docs/reference/proto3-spec
```
Letters and digits  
  letter = "A" … "Z" | "a" … "z"
  decimalDigit = "0" … "9"

Identifiers
  ident = letter { letter | decimalDigit | "_" }
  fieldName = ident
```

It shouldn't be a problem. There's still a bug, though, for Protobuf  before and after this PR where CREATE STREAM does  not accept `.` column names (https://github.com/confluentinc/ksql/issues/9005). I am unsure if we should allow that yet, but this PR does not fix it.

For the other logic, such as Maps. This is the part I agree I don't know if we need it. I will figure out if I can get rid of it for Protobuf.